src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.TryCloneFile.OSX.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+namespace System.IO
+{
+    internal static partial class FileSystem
+    {
+        private static bool TryCloneFile(string sourceFullPath, string destFullPath, bool overwrite)
+        {
+            // This helper function calls out to clonefile, and returns the error.
+            static bool TryCloneFile(string sourceFullPath, string destFullPath, int flags, out Interop.Error error)
+            {
+                if (Interop.@libc.clonefile(sourceFullPath, destFullPath, flags) == 0)
+                {
+                    // Success.
+                    error = Interop.Error.SUCCESS;
+                    return true;
+                }
+                error = Interop.Sys.GetLastError();
+                return false;
+            }
+
+            // Try to clone the file immediately, this will only succeed if the
+            // destination doesn't exist, so we don't worry about locking for this one.
+            int flags = Interop.@libc.CLONE_ACL;
+            Interop.Error error;
+            if (TryCloneFile(sourceFullPath, destFullPath, flags, out error)) return true;
+
+            // Some filesystems don't support ACLs, so may fail due to trying to copy ACLs.
+            // This will disable them and allow trying again (a maximum of 1 time).
+            if (error == Interop.Error.EINVAL)
+            {
+                flags = 0;
+                if (TryCloneFile(sourceFullPath, destFullPath, flags, out error)) return true;
+            }
+
+            // Try to delete the destination file if we're overwriting.
+            if (error == Interop.Error.EEXIST && overwrite)
```

Comment:
```suggestion
            else if (error == Interop.Error.EEXIST && overwrite) // Try to delete the destination file if we're overwriting.
```

Reply:
This can't be `else if` since `error` can be changed in the previous `if` block.